internal/provider/provider.go

```diff
@@ -0,0 +1,32 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+)
+
+func New() tfsdk.Provider {
+	return &provider{}
+}
+
+var _ tfsdk.Provider = (*provider)(nil)
+
+type provider struct{}
+
+func (p provider) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
```

Comment:
Typically pointer receivers are used for functions defined on types implementing the various interfaces defined in `terraform-plugin-framework` (e.g., `tfsdk.Provider`, `tfsdk.Resource`). There is an example of this in the [online tutorial](https://learn.hashicorp.com/tutorials/terraform/plugin-framework-create?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS#explore-the-hashicups-provider).